frontend/src/components/dashboard/create/telegram/TelegramCredentials.tsx

```diff
@@ -21,14 +28,15 @@ import TelegramCredentialsInput from './TelegramCredentialsInput'
 import TelegramValidationInput from './TelegramValidationInput'
 import styles from '../Create.module.scss'
 import { i18n } from 'locales'
+import { TelegramCampaign, TelegramProgress } from 'classes'
 
 const TelegramCredentials = ({
-  hasCredential: initialHasCredential,
-  onNext,
+  setActiveStep,
 }: {
-  hasCredential: boolean
-  onNext: (changes: any, next?: boolean) => void
+  setActiveStep: Dispatch<SetStateAction<TelegramProgress>>
 }) => {
+  const { campaign, setCampaign } = useContext(CampaignContext)
+  const { hasCredential: initialHasCredential, progress } = campaign
```

Comment:
```suggestion
  const { hasCredential: initialHasCredential } = campaign
```

Minor issue, I think `progress` is unused.